docs/dev/release_checklist.md

```diff
@@ -88,32 +84,45 @@ Please see the [official Poetry documentation on `version`](https://python-poetr
 !!! important
     The changelog must adhere to the [Keep a Changelog](https://keepachangelog.com/) style guide.
 
-This example uses 1.4.2, but change the version number to match the version you bumped to in the previous step. First, create a release branch off of `develop` (`git switch -c release-1.4.2 develop`).
+This guide uses `1.4.2` as the new version in its examples, so change it to match the version you bumped to in the previous step! Every. single. time. you. copy/paste commands :)
+
+First, create a release branch off of `develop` (`git switch -c release-1.4.2 develop`).
+
+> You will need to have the project's poetry environment built at this stage, as the towncrier command runs **locally only**. If you don't have it, run `poetry install` first.
+
+Generate release notes with `invoke generate-release-notes --version 1.4.2` and answer `yes` to the prompt `Is it okay if I remove those files? [Y/n]:`. This will update the release notes in `docs/admin/release_notes/version_X.Y.md`, stage that file in git, and `git rm` all the fragments that have now been incorporated into the release notes.
 
-Generate release notes with `invoke generate-release-notes --version 1.4.2` and answer `yes` to the prompt `Is it okay if I remove those files? [Y/n]:`. This will update the release notes in `docs/admin/release_notes/version_1.4.md`, stage that file in git, and `git rm` all the fragments that have now been incorporated into the release notes.
+There are two possibilities:
 
-Check the git diff to verify the changes are correct (`git diff --cached`).
+1. If you're releasing a new major or minor version, rename the `version_X.Y.md` file accordingly (e.g. rename to `docs/admin/release_notes/version_1.4.md`). Update the `Release Overview` and add this new page to the table of contents within `mkdocs.yml`.
+2. If you're releasing a patch version, copy your version's section from the `version_X.Y.md` file into the already existing `docs/admin/release_notes/version_1.4.md` file. Delete the `version_X.Y.md` file.
 
-Commit and push the staged changes.
+Check the `git diff` to verify the changes are correct (`git diff --cached`). Remember to also stage `git add pyproject.toml`.
+
+Commit `git commit -m "Create release v1.4.2"` and push the staged changes.
 
 ### Submit Release Pull Request
 
-Submit a pull request to merge your release branch into `develop`. Once merged, submit another pull request titled `**"Release vX.Y.Z"**` to merge the `develop` branch into `main`. Copy the documented release notes into the pull request's body.
+Submit a pull request titled `**"Release vX.Y.Z"**` to merge your release branch into `develop`.
 
 Once CI has completed on the PR, merge it.
 
+Once merged into `develop`, submit another pull request with the same title to merge `develop` into `main`. Copy the documented release notes into the pull request's body.
+
```

Comment:
FWIW this is different than the process we use in nautobot core - there the release branch is created from develop and PRed directly into main, rather than doing two separate release PRs.

Reply:
Yeah I'm not entirely sure what's the point of release->develop->main ... perhaps @smk4664 knows since this is based on his edits on the core checklist.

Reply:
Since most apps were doing release->main->develop originally, I'll simplify this for now.

Reply:
Thanks both! Instructions updated accordingly.